plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/preferences/OpenShiftCorePreferences.java

```diff
@@ -61,7 +63,11 @@ private StringPreferenceValue createPreferenceValue(String connectionURL) {
 	}
 	
 	public String getOCBinaryLocation() {
-		return ocBinaryLocation.get();
+		String result = ocBinaryLocation.get();
+		if(StringUtils.isEmpty(result)) {
```

Comment:
That is ok, just as I wrote in the other PR, OpenShiftPreferencePage and OpenShiftCorePreferenceInitializer need  method in OCBinary(Name) that would work as before to avoid cycle to preferences instance scope.